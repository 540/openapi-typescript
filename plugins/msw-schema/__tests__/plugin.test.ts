import { generateCode, loadFixture } from "./utils.js";

describe("msw-schema plugin", () => {
  test("should generate correct TypeScript interfaces for empty OpenAPI spec", async () => {
    const spec = loadFixture("minimal-openapi-no-paths.json");

    const generatedCode = await generateCode(spec);

    expect(generatedCode).toMatchSnapshot();
  });

  test("should generate correct TypeScript interfaces for basic API endpoints", async () => {
    const spec = loadFixture("simple-path-201-no-body.json");

    const generatedCode = await generateCode(spec);

    expect(generatedCode).toMatchSnapshot();
  });

  test("should generate correct TypeScript interfaces for endpoints with request bodies", async () => {
    const spec = loadFixture("path-with-request-body.json");

    const generatedCode = await generateCode(spec);

    expect(generatedCode).toMatchSnapshot();
  });

  test("should generate correct TypeScript interfaces for endpoints with complex responses", async () => {
    const spec = loadFixture("path-with-200-response-object.json");

    const generatedCode = await generateCode(spec);

    expect(generatedCode).toMatchSnapshot();
  });
});
