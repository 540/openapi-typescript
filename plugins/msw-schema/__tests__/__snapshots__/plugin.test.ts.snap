// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`msw-schema plugin should generate correct TypeScript interfaces for basic API endpoints 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export interface paths {
    "/resource": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        get?: never;
        put?: never;
        /** Create a new resource */
        post: operations["createResource"];
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
export type webhooks = Record<string, never>;
export interface components {
    schemas: never;
    responses: never;
    parameters: never;
    requestBodies: never;
    headers: never;
    pathItems: never;
}
export type $defs = Record<string, never>;
export interface operations {
    createResource: {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Resource created successfully */
            201: {
                headers: {
                    [name: string]: unknown;
                };
                content?: never;
            };
        };
    };
}
"
`;

exports[`msw-schema plugin should generate correct TypeScript interfaces for empty OpenAPI spec 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export type paths = Record<string, never>;
export type webhooks = Record<string, never>;
export interface components {
    schemas: never;
    responses: never;
    parameters: never;
    requestBodies: never;
    headers: never;
    pathItems: never;
}
export type $defs = Record<string, never>;
export type operations = Record<string, never>;
"
`;

exports[`msw-schema plugin should generate correct TypeScript interfaces for endpoints with complex responses 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export interface paths {
    "/resource": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        /** Get a resource */
        get: operations["getResource"];
        put?: never;
        post?: never;
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
export type webhooks = Record<string, never>;
export interface components {
    schemas: never;
    responses: never;
    parameters: never;
    requestBodies: never;
    headers: never;
    pathItems: never;
}
export type $defs = Record<string, never>;
export interface operations {
    getResource: {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Resource retrieved successfully */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": {
                        id?: string;
                        name?: string;
                    };
                };
            };
        };
    };
}
"
`;

exports[`msw-schema plugin should generate correct TypeScript interfaces for endpoints with request bodies 1`] = `
"import { createOpenApiHttp as baseCreateOpenApiHttp } from "openapi-msw";
export const createOpenApiHttp: typeof baseCreateOpenApiHttp<paths> = baseCreateOpenApiHttp;
export interface paths {
    "/resource": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        get?: never;
        put?: never;
        /** Create a resource with request body */
        post: operations["createResourceWithBody"];
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
export type webhooks = Record<string, never>;
export interface components {
    schemas: never;
    responses: never;
    parameters: never;
    requestBodies: never;
    headers: never;
    pathItems: never;
}
export type $defs = Record<string, never>;
export interface operations {
    createResourceWithBody: {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody: {
            content: {
                "application/json": {
                    name: string;
                    description?: string;
                };
            };
        };
        responses: {
            /** @description Resource created successfully */
            201: {
                headers: {
                    [name: string]: unknown;
                };
                content?: never;
            };
        };
    };
}
"
`;
